.github/workflows/build.yaml

```diff
@@ -0,0 +1,115 @@
+name: Build
+
+on:
+  workflow_call: 
+    inputs:
+      version:
+        type: string
+        description: |
+          The version of the project to build. Example: `1.0.3`.
+          
+          If not provided, a development build with a version name
+          based on the branch name will be built. Otherwise, a release
+          build with the provided version will be built.
+        required: false
+
+jobs:
+  # phase 1
+  target:
+    name: Build target branch
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    outputs:
+      target_branch: ${{ steps.build-target.outputs.target_branch }}
+      version: ${{ steps.build-target.outputs.version }}
+      docker_matrix: ${{ steps.build-target.outputs.docker_matrix }}
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+
+      - name: Install Viash
+        uses: viash-io/viash-actions/setup@v6
+      
+      - name: Determine variables
+        id: variables
+        run: |
+          VERSION="${{ github.event.inputs.version }}"
+          SOURCE_BRANCH=$(echo $GITHUB_REF | sed 's/refs\/heads\///')
+
+          if [[ -z $VERSION ]]; then
+            TARGET_BRANCH="build/$SOURCE_BRANCH"
+            VERSION=$(echo $TARGET_BRANCH | sed 's/[^a-zA-Z0-9_]/_/g')
```

Comment:
Might be a good idea to quote here because `$TARGET_BRANCH` could (but should not) contain any weird characters

Reply:
Done!